invenio_drafts_resources/services/draft.py

```diff
@@ -27,57 +29,114 @@ class RecordDraftServiceConfig(RecordServiceConfig):
 
     # RecordService configuration
     resource_unit_cls = IdentifiedRecordDraft
-    data_validator = MarshmallowDataValidator(
-        schema=DraftMetadataSchemaJSONV1
-    )
 
     # DraftService configuration.
     # WHY: We want to force user input choice here.
     draft_cls = None
+    draft_data_validator = MarshmallowDataValidator(
+        schema=DraftMetadataSchemaJSONV1
+    )
 
 
 class RecordDraftService(RecordService):
     """Draft Service interface."""
 
+    RecordIdProviderV2.default_status_with_obj = PIDStatus.NEW
     default_config = RecordDraftServiceConfig
 
+    # Over write resolver because default status is a class attr
+    @property
+    def resolver(self):
+        """Factory for creating a draft resolver instance."""
+        return self.config.resolver_cls(
+            pid_type=self.config.resolver_pid_type,
+            getter=self.record_cls.get_record,
+            registered_only=False
+        )
+
+    # Draft attrs
+    @property
+    def draft_cls(self):
+        """Factory for creating a record class."""
+        return self.config.draft_cls
+
+    @property
+    def draft_data_validator(self):
+        """Returns an instance of the draft data validator."""
+        return self.config.draft_data_validator
+
+    @property
+    def draft_resolver(self):
+        """Factory for creating a draft resolver instance."""
+        return self.config.resolver_cls(
+            pid_type=self.config.resolver_pid_type,
+            getter=self.draft_cls.get_record,
+            registered_only=False
+        )
+
+    def resolve_draft(self, id_):
+        """Resolve a persistent identifier to a record draft."""
+        return self.draft_resolver.resolve(id_)
+
     # High-level API
     # Inherits record read, search, create, delete and update
 
-    def _index_draft(self, draft):
-        indexer = self.indexer()
-        if indexer:
-            indexer.index(draft)
+    def read_draft(self, id_, identity):
```

Comment:
normal: we can keep it named "read" , right? It operates on a draft and all the methods below as well except for the `edit` method. It could be renamed `edit_record`.